Packs/GigamonThreatINSIGHT/Integrations/GigamonThreatINSIGHT/GigamonThreatINSIGHT.py

```diff
@@ -400,6 +401,80 @@ def formatEvents(r_json):
     return newData
 
 
+def getFirstFetch(firstFetch) -> dict[str, Any]:
+
+    pieces = firstFetch.strip().split()
+    unit = pieces[len(pieces) - 1]
+
+    try:
+        delta = arg_to_number(
+            arg=pieces[0],
+            arg_name='first_fetch',
+            required=False
+        )
+    except Exception:
+        delta = 0
+
+    days: int = delta if delta and unit == 'days' else 0
+    hours: int = delta if delta and unit == 'hours' else 0
+
+    if days == 0 and hours == 0:
+        days = 7
+
+    return {'days': days, 'hours': hours}
+
+
+def mapSeverity(severity) -> int:
+    match severity:
+        case 'high':
+            return 3
+        case 'moderate':
+            return 2
+        case 'low':
+            return 1
+        case _:
+            return 0
+
+
+def getIncidents(result, last_fetch) -> Tuple[Dict[str, int], List[dict[str, Any]]]:
+    # Initialize an empty list of incidents to return
+    # Each incident is a dict with a string as a key
+    incidents: List[Dict[str, Any]] = []
+
+    last_incident_time = last_fetch
+    for detection in result.outputs:
+        incident_time = datetime.strptime(detection['first_seen'], DATE_FORMAT)
+
+        # Check if inciden has been reported before
+        if last_fetch >= incident_time:
+            continue
+
+        severity = mapSeverity(detection['rule_severity'])
+
+        incident = {
+            'name': detection['rule_name'],
+            'occurred': detection['first_seen'],
+            'severity': severity,
+            'details': detection['rule_description'],
+            'dbotMirrorId': detection['uuid'],
+            'rawJSON': json.dumps(detection),
+        }
+
+        incidents.append(incident)
+
+        if last_incident_time < incident_time:
+            last_incident_time = incident_time
+
+    demisto.debug(
+        f'Last incident time: {last_incident_time.strftime(DATE_FORMAT)}')
+
+    next_run = {'last_fetch': last_incident_time.strftime(DATE_FORMAT)}
```

Comment:
Should last_fetch be the system time the fetch command was executed and not related to a detection?

Reply:
The question with this is that we filter detections by this date. would it be possible we receive detections delayed? I mean, if we make the fetch at time t, could we receive a detection with a time stamp before t later on? I think it would be possible some delay in which case we should not use system time.